src/graphql/graphql-api.php

```diff
@@ -0,0 +1,255 @@
+
+namespace WPCOMVIP\BlockDataApi;
+
+use GraphQLRelay\Relay;
+
+defined( 'ABSPATH' ) || die();
+
+/**
+ * GraphQL API to offer an alternative to the REST API.
+ */
+class GraphQLApi {
+	/**
+	 * Initiatilize the graphQL API by hooking into the graphql_register_types action, 
+	 * which only fires if WPGraphQL is installed and enabled, and is further controlled 
+	 * by the vip_block_data_api__is_graphql_enabled filter.
+	 */
+	public static function init() {
+		add_action( 'graphql_register_types', [ __CLASS__, 'register_types' ] );
+	}
+
+	/**
+	 * Extract the blocks data for a post, and return back in the format expected by the graphQL API.
+	 *
+	 * @param  \WPGraphQL\Model\Post $post_model Post model for post.
+	 * 
+	 * @return array
+	 */
+	public static function get_blocks_data( $post_model ) {
+		$post_id = $post_model->ID;
+		$post    = get_post( $post_id );
+
+		$content_parser = new ContentParser();
+
+		$parser_results = $content_parser->parse( $post->post_content, $post_id );
+
+		// We need to not return a WP_Error object, and so a regular exception is returned.
+		if ( is_wp_error( $parser_results ) ) {
+			Analytics::record_error( $parser_results );
+
+			// Return API-safe error with extra data (e.g. stack trace) removed.
+			return new \Exception( $parser_results->get_error_message() );
+		}
+
+		$parser_results['blocks'] = array_map(function ( $block ) {
+			return self::transform_block_format( $block );
+		}, $parser_results['blocks'] );
+
+		return $parser_results;
+	}
+
+	/**
+	 * Transform the block's format to the format expected by the graphQL API.
+	 *
+	 * @param array $block An associative array of parsed block data with keys 'name' and 'attribute'.
+	 *
+	 * @return array
+	 */
+	public static function transform_block_format( $block ) {
+		// Generate a unique ID for the block.
+		$block['id'] = Relay::toGlobalId( 'ID', wp_unique_id() );
+
+		// Convert the attributes to be in the name-value format that the schema expects.
+		$block = self::map_attributes( $block );
+
+		if ( isset( $block['innerBlocks'] ) && ! empty( $block['innerBlocks'] ) ) {
+			$block['innerBlocks'] = array_map( function ( $block ) {
+				return self::transform_block_format( $block );
+			}, $block['innerBlocks'] );
+		}
+
+		return $block;
+	}
+
+	/**
+	 * Convert the attributes to be in the name-value format that the schema expects.
+	 *
+	 * @param array $block An associative array of parsed block data with keys 'name' and 'attribute'.
+	 *
+	 * @return array
+	 */
+	public static function map_attributes( $block ) {
+		// check if type of attributes is stdClass and unset it as that's not supported by graphQL.
+		if ( isset( $block['attributes'] ) && is_object( $block['attributes'] ) ) {
+			unset( $block['attributes'] );
+		} elseif ( isset( $block['attributes'] ) && ! empty( $block['attributes'] ) ) {
+			$block['attributes'] = array_map(
+				function ( $name, $value ) {
+					return [
+						'name'  => $name,
+						'value' => $value,
```

Comment:
I guess WPGraphQL is doing the casting to string? You might want to do it here in code just to be explicit and test it.

Reply:
> I guess WPGraphQL is doing the casting to string?

I think you're right! I tested with a block delimiter value like `{"test": 123}`, which is returned as an integer from the parser and REST API. However, once we return it in a `BlockAttribute`, it is converted to a string. Added explicit conversion in https://github.com/Automattic/vip-block-data-api/pull/51/commits/8cdc1e209fb2688dc0930e53ba1e5f0733da18b5 anyway.